server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -0,0 +1,21 @@
+package org.apache.james.jmap.http
+
+import javax.inject.Inject
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import reactor.core.publisher.Mono
+import reactor.core.scala.publisher.SFlux
+import reactor.netty.http.server.HttpServerRequest
+import scala.compat.java8.StreamConverters._
+class BasicAuthenticationStrategy @Inject() (val basicAuthTokenManager: BasicAuthenticationTokenManager,
```

Comment:
The BasicAuthenticationStrategy is highly coupled with the `BasicAuthenticationTokenManager` class.

I do believe getting two classes here makes little sense. Can we group them into one?